src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -460,5 +463,43 @@ public void ReleaseSubscriptionsSemaphore()
         {
             _concurrentConnectionsSemiSemaphore.Release();
         }
+
+        internal void CleanupSubscriptions()
+        {
+            var maxTaskLifeTime = (PlatformDetails.Is32Bits || this._db.Configuration.Storage.ForceUsing32BitsPager
+                        ? TimeSpan.FromHours(12)
+                        : TimeSpan.FromDays(2));
+            var oldestPossibleIdleSubscription = SystemTime.UtcNow - maxTaskLifeTime;
+            foreach (var state in _subscriptionConnectionStates)
+            {
+                if (state.Value.Connection != null)
+                    continue;
+
+                var recentConnection = state.Value.MostRecentEndedConnection();
+                if (recentConnection == null)
+                    return;
+
+                if (recentConnection.Stats.LastMessageSentAt < oldestPossibleIdleSubscription)
```

Comment:
You are removing just one here, what if we have a lot of them? Should't we put this in a while?

Reply:
done